litmus-portal/frontend/src/pages/WorkflowUnderground/index.tsx

```diff
@@ -1,96 +1,60 @@
-import React, { useState, useEffect } from 'react';
+/* eslint-disable camelcase */
+
 import { useSubscription } from '@apollo/client';
 import { Typography } from '@material-ui/core';
-import Scaffold from '../../containers/layouts/Scaffold';
+import React, { useEffect, useState } from 'react';
 import SideBar from '../../components/Sections/WorkflowUnderground/WorkflowRepresentation';
-import useStyles from './styles';
-import Loader from '../../components/Loader';
+import Scaffold from '../../containers/layouts/Scaffold';
 import { WORKFLOW_EVENTS } from '../../graphql';
+import { LocationState } from '../../models/routerModel';
+import {
+  ExecutionData,
+  WorkflowRun,
+  WorkflowSubscription,
+} from '../../models/workflowData';
+import capitalize from '../../utils/capitalize';
+import useStyles from './styles';
 
 interface WorkflowUndergroundProps {
-  location: any;
+  location: LocationState<WorkflowRun>;
 }
 
 const WorkflowUnderground: React.FC<WorkflowUndergroundProps> = ({
   location,
 }) => {
   const classes = useStyles();
+  const [data, setData] = useState<WorkflowRun>(location.state);
 
-  const [isInitial, setIsInitial] = useState<boolean>(true);
-  const [data, setData] = useState<any>({
-    workflow_name: '',
-    workflow_run_id: '',
-    execution_data: '',
-    last_updated: '',
-    cluster_name: '',
-  });
-
-  const dataSub = useSubscription(WORKFLOW_EVENTS);
+  const dataSub = useSubscription<WorkflowSubscription>(WORKFLOW_EVENTS);
```

Comment:
I think the previous PR added variables to the subscription...can you use them here...we have a WorkflowVars interface in models folder

Reply:
Updated :heavy_check_mark: